application/app/Models/Role.php

```diff
@@ -69,11 +72,17 @@ public function privilegeRoles(): HasMany
 
     public function privileges(): BelongsToMany
     {
-        return $this->belongsToMany(Privilege::class, PrivilegeRole::class);
+        return $this->belongsToMany(Privilege::class, PrivilegeRole::class)
+            ->using(PrivilegeRole::class);
     }
 
     public function institution(): BelongsTo
     {
         return $this->belongsTo(Institution::class);
     }
+
+    public function institutionUsers(): BelongsToMany
+    {
+        return $this->belongsToMany(InstitutionUser::class, InstitutionUserRole::class);
+    }
```

Comment:
**[blocker]** Soft-deleted pivots are being included.

![image](https://github.com/keeleinstituut/tv-authorization/assets/26142658/bfe9d8c4-ee8f-417a-8c7f-2485a46b5adf)

```suggestion
    public function institutionUsers(): BelongsToMany
    {
        return $this->belongsToMany(InstitutionUser::class, InstitutionUserRole::class)
            ->wherePivot('deleted_at', null);
    }
```

Reply:
This has been discussed before (sync + pivot model + softdeletes), Laravel can't handle this well for some reason. This is exact change will yield to unpredictable cases when saving the objects

Reply:
@thenouan I understand your point about soft-deletion and Laravel’s `BelongsToMany` not playing well together. However, the "unpredictable cases" problem is already here: see my example above. In my personal opinion it is not good practice to allow such unexpected behaviour into the codebase.

I see two solutions: 
* Remove the `SoftDeletes` trait from the `InstitutionUserRole` class.
* Remove this method since the `BelongsToMany` can’t be relied upon. Instead use chained access via `Role`->`InstitutionUserRole`->`InstitutionUser`.